lifecycle.md

```diff
@@ -0,0 +1,54 @@
+# Prereq container image lifecycle
+
+The container images in this repository are provided for building .NET source and testing the resultant binaries. The images are built with OS base images and must be updated per the OS lifecycle. This document describes the required patterns.
+
+Container images are produced for each of the supported Linux distros listed in [Supported OS Versions](https://github.com/dotnet/core/blob/main/os-lifecycle-policy.md) files.
```

Comment:
They are produced on an as needed basis.  Not all supported distro's especially the cross product of supported architectures will exist unless there is a need for them.

Reply:
I think the truth is in-between. Ignoring architectures, which distros do you see that are skipped? It appears to me that we've conflated SUSE and OpenSUSE into one, but otherwise all distros are represented. I can change the wording, however, it is good to get on the same page first.

Reply:
`which distros do you see that are skipped` - Red Hat.

The similar statement in the test section will also need updating:

`Test images are provided for each of the supported OS versions. They are built for each supported architecture.`

Reply:
We have CentOS Stream and Alma Linux images to cover for Red Hat support.

That's covered by this statement: https://github.com/dotnet/core/blob/main/support.md#compatible-derivatives. I can add a link to that.

Reply:
I'll address the supported architectures. As you suggest, we only build for all architectures in some cases.